sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -72,10 +43,9 @@ public final class RestProxy implements InvocationHandler {
      * this RestProxy "implements".
      */
     private RestProxy(HttpPipeline httpPipeline, SerializerAdapter serializer, SwaggerInterfaceParser interfaceParser) {
-        this.httpPipeline = httpPipeline;
-        this.serializer = serializer;
         this.interfaceParser = interfaceParser;
-        this.decoder = new HttpResponseDecoder(this.serializer);
+        this.asyncRestProxy = new AsyncRestProxy(httpPipeline, serializer, interfaceParser);
+        this.syncRestProxy = new SyncRestProxy(httpPipeline, serializer, interfaceParser);
```

Comment:
we might not need both at the same time ?

Reply:
The service might look like 
```
interface Service {
   void doFoo();
  Mono<Void> doFooAsync();
}
```

IMO, trying to detect how the interface look like to spare that allocation is not worth it. (Especially that these two may merge eventually).